hudi-flink-datasource/hudi-flink1.14.x/src/main/java/org/apache/hudi/table/format/cow/vector/reader/ParquetColumnarRowSplitReader.java

```diff
@@ -218,11 +218,17 @@ private WritableColumnVector[] createWritableVectors() {
     List<Type> types = requestedSchema.getFields();
     List<ColumnDescriptor> descriptors = requestedSchema.getColumns();
     for (int i = 0; i < requestedTypes.length; i++) {
-      columns[i] = createWritableColumnVector(
-          batchSize,
-          requestedTypes[i],
-          types.get(i),
-          descriptors);
+      String fieldName = requestedSchema.getFieldName(i);
```

Comment:
Is it possible to move this line to the catch block?

Reply:
Hi @danny0405 ,

Correct. It should be moved to the catch block as it would be only needed if there was an exception.